scanner/src/main/java/ca/on/oicr/gsi/runscanner/scanner/Scheduler.java

```diff
@@ -361,15 +385,18 @@ public boolean isScanningNow() {
   /**
    * Determine if a run directory is in need of processing.
    *
-   * <p>This means that it is not in a processing queue, failed processing last time, nor needs
-   * reprocessing (for runs still active on the sequencer)
+   * <p>This means that it is not in a processing queue, failed processing last time, not in the
+   * list of subdirectories to ignore nor needs reprocessing (for runs still active on the
+   * sequencer)
    */
   private boolean isUnprocessed(File directory) {
     return !workToDo.contains(directory)
         && !processing.contains(directory)
         && (!failed.containsKey(directory)
             || Duration.between(failed.get(directory), Instant.now()).toMinutes() > 20)
-        && (!finishedWork.containsKey(directory) || finishedWork.get(directory).shouldRerun());
+        && (!finishedWork.containsKey(directory) || finishedWork.get(directory).shouldRerun())
+        // Exclude from processing if directory name in list of directories to ignore
+        && roots.stream().noneMatch(n -> n.getIgnoreSubdirectories().contains(directory.getName()));
```

Comment:
Can't build runscanner to test right now - isUnprocessed is called as a filter while streaming all the roots (which is a List\<Configuration\>), this method also streams all the roots, and it doesn't look like ignoreSubdirectories is storing the absolute path. Is there interference across Configurations? If, for example, 'test' is listed as an ignoredSubdirectory for a Miseq, would it also get ignored for a NovaSeq (held in a different configuration)?

Reply:
right, so for the following configuration file:
```
[
  {
    "name": "nextseq",
    "parameters": {},
    "path": "/home/avarsava/nextseq/",
    "platformType": "ILLUMINA",
    "timeZone": "America/Toronto",
    "ignoreSubdirectories": [
      "RECYCLE"
    ]
  },
  {
    "name": "miseq",
    "parameters": {},
    "path": "/home/avarsava/miseq/",
    "platformType": "ILLUMINA",
    "timeZone": "America/Toronto",
    "ignoreSubdirectories": []
  }
]
```
This will create 2 Configuration objects which are streamed by 'roots.stream()'. `/home/avarsava/miseq/RECYCLE' would match by this logic even though it's not specified for exclusion, would it not?

Reply:
Yep. It's currently checking the `ignoreSubdirectories` for every root instead of only the specific root that it belongs to